xarray/core/groupby.py

```diff
@@ -1526,7 +1526,14 @@ def _combine(self, applied, shortcut=False):
         if shortcut:
             combined = self._concat_shortcut(applied, dim, positions)
         else:
-            combined = concat(applied, dim)
+            combined = concat(
+                applied,
+                dim,
+                data_vars="all",
+                coords="different",
+                compat="equals",
+                join="outer",
+            )
```

Comment:
I hard-coded these to the old defaults since there is no way for the user to set them.